File: 1557-minimum-number-of-vertices-to-reach-all-nodes/1557-minimum-number-of-vertices-to-reach-all-nodes.cpp
class Solution {
public:
    vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) {
        vector<bool> isIncomingEdgeExists(n, false);
        for(auto edge : edges) isIncomingEdgeExists[edge[1]] = true;
        vector<int> ans;
        for(int i=0; i<n; i++){
            if(!isIncomingEdgeExists[i]) ans.push_back(i);
        }
        return ans;
    }
};

File: 1557-minimum-number-of-vertices-to-reach-all-nodes/1557-minimum-number-of-vertices-to-reach-all-nodes.py
class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: List[List[int]]) -> List[int]:
        inDeg = [0]*n
        for e in edges: inDeg[e[1]]+=1
        ans = []
        for i in range(0, n):
            if inDeg[i]==0: ans.append(i)
        return ans